source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -445,6 +445,13 @@ void FaultFilter::postDelayInjection(const Http::RequestHeaderMap& request_heade
   if (http_status.has_value()) {
     abortWithStatus(http_status.value(), grpc_status);
   } else {
+    // Should not continue to count as an active fault after the delay has elapsed
+    // if no other type of fault is active/injected
+    if (response_limiter_ == nullptr && fault_active_) {
```

Comment:
I don't think this guarantees there's no active fault processing, couldn't there be a response rate limiter still applying?  Maybe have a activeFaults() helper function which checks for delay timer and response_rate_limit_, and only decrements if they're both gone?

Reply:
Thanks for the comments. Added with a new helper to check both delay timer and response_rate_limit_.